src/DotNetty.Common/Internal/PlatformDependent.cs

```diff
@@ -53,17 +53,15 @@ public static unsafe void CopyMemory(byte[] src, int srcIndex, byte[] dst, int d
         {
             if (length > 0)
             {
-                fixed (byte* source = &src[srcIndex])
-                    fixed (byte* destination = &dst[dstIndex])
-                        Unsafe.CopyBlock(destination, source, unchecked((uint)length));
+                Unsafe.CopyBlockUnaligned(ref dst[dstIndex], ref src[srcIndex], unchecked((uint)length));
```

Comment:
Unsafe does a buffer pin/unpin, for unsafe buffers (already pinned on the arena), this is extra performance overhead
You have to something like
                IntPtr ptr = dst.AddressOfPinnedMemory();
                if (ptr != IntPtr.Zero)
                {
                    PlatformDependent.CopyMemory(addr, (byte*)(ptr + dstIndex), length);
                }
                else
                {
                    PlatformDependent.CopyMemory(addr, destination + dstIndex, length);
                    fixed (byte* destination = &dst.GetPinnableMemoryAddress())
                    {
                        PlatformDependent.CopyMemory(addr, destination + dstIndex, length);
                    }
                }
That is the reason why PlatformDependent does not use the ref param methods.

Reply:
I didn't recognized it's already pinned in `DirectArena.MemoryChunk` before.

But I think maybe ref is no need to be pinned, and this commit didn't change any overload with an pointer as param to ref. And the only thing need to be change is `UnsafeByteBufferUtil.Copy`, which is out of scope of this PR.

Maybe I will do some test to see the difference latter.

Reply:
@yyjdelete ref does a pin/unpin (fixed) for sure, your changes are ok as is. I plan to consolidate all unsafe calls.

Reply:
@StormHub so this is good with you?